CRM/Contribute/BAO/Contribution.php

```diff
@@ -90,6 +90,8 @@ public static function add(&$params, $ids = []) {
     if (empty($params)) {
       return NULL;
     }
+    $contributionID = CRM_Utils_Array::value('contribution', $ids, CRM_Utils_Array::value('id', $params));
```

Comment:
isn't this duplicating L96 and if so should the comment be moved up above this line and L96 removed?

Reply:
yep